src/mesh/rectilineargrid.jl

```diff
@@ -35,12 +35,13 @@ vertex(g::RectilinearGrid{Dim}, ijk::Dims{Dim}) where {Dim} = Point(getindex.(g.
 
 xyz(g::RectilinearGrid) = g.xyz
 
-@generated function XYZ(g::RectilinearGrid{Dim,T}) where {Dim,T}
+XYZ(g::RectilinearGrid{Dim,T}) where {Dim,T} = xyzXYZ(T, xyz(g))
+
+@generated function xyzXYZ(::Type{T}, xyz::NTuple{Dim}) where {T,Dim}
   exprs = ntuple(Dim) do d
     quote
-      a = g.xyz[$d]
-      N = length(a)
-      A = Array{T,Dim}(undef, @ntuple($Dim, i -> N))
+      A = Array{T,Dim}(undef, length.(xyz))
+      a = xyz[$d]
```

Comment:
This order looks better visually:
```suggestion
      a = xyz[$d]
      A = Array{T,Dim}(undef, length.(xyz))
```